datalad/cmdline/main.py

```diff
@@ -252,23 +253,31 @@ def main(args=None):
         args_ = strip_arg_from_argv(args or sys.argv, cmdlineargs.pbs_runner, pbs_runner_opt[1])
         # run the function associated with the selected command
         run_via_pbs(args_, cmdlineargs.pbs_runner)
-    elif cmdlineargs.common_debug or cmdlineargs.common_idebug:
-        # so we could see/stop clearly at the point of failure
-        setup_exceptionhook(ipython=cmdlineargs.common_idebug)
-        ret = cmdlineargs.func(cmdlineargs)
-    else:
-        # otherwise - guard and only log the summary. Postmortem is not
-        # as convenient if being caught in this ultimate except
-        try:
+    elif has_func:
+        if cmdlineargs.common_debug or cmdlineargs.common_idebug:
+            # so we could see/stop clearly at the point of failure
+            setup_exceptionhook(ipython=cmdlineargs.common_idebug)
             ret = cmdlineargs.func(cmdlineargs)
-        except InsufficientArgumentsError as exc:
-            # if the func reports inappropriate usage, give help output
-            lgr.error('%s (%s)' % (exc_str(exc), exc.__class__.__name__))
-            cmdlineargs.subparser.print_usage()
-            sys.exit(1)
-        except Exception as exc:
-            lgr.error('%s (%s)' % (exc_str(exc), exc.__class__.__name__))
-            sys.exit(1)
+        else:
+            # otherwise - guard and only log the summary. Postmortem is not
+            # as convenient if being caught in this ultimate except
+            try:
+                ret = cmdlineargs.func(cmdlineargs)
+            except InsufficientArgumentsError as exc:
+                # if the func reports inappropriate usage, give help output
+                lgr.error('%s (%s)' % (exc_str(exc), exc.__class__.__name__))
+                cmdlineargs.subparser.print_usage()
+                sys.exit(1)
+            except Exception as exc:
```

Comment:
that is strange that codecov reports these lines uncovered, although travis shows running the test and local run reporting those lines covered:
```shell
(git)hopa:~datalad/datalad[bf-mispecified]git
$> rm -f .coverage; python -m nose -s -v --with-cov --cover-package datalad datalad/cmdline/tests/test_main.py >/dev/null 2>&1 | grep cmdline/main                    
datalad/cmdline/main.py                      116     40    66%   140, 205, 227-228, 232-282
```